### PR TITLE
refine multi-backend setup.py

### PR DESCRIPTION
**Pull request by jinminxi104**
refine multi-backend setup.py

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docker/Dockerfile_aarch64_ascend [hunk 1]
```diff
@@ -122,4 +122,4 @@ WORKDIR /opt/lmdeploy
 
 RUN --mount=type=cache,target=/root/.cache/pip \
     sed -i '/triton/d' requirements/runtime.txt && \
-    pip3 install -v --no-build-isolation -e .
+    LMDEPLOY_BACKEND=ascend pip3 install -v --no-build-isolation -e .
```
Comment (lvhan028) [inline docker/Dockerfile_aarch64_ascend:125]: How about LMDEPLOY_TARGET_DEVICE?

File: setup.py [hunk 2]
```diff
@@ -4,18 +4,14 @@
...
-
-    npu_available = torch_npu.npu.is_available()
-except ImportError:
-    pass
-
 pwd = os.path.dirname(__file__)
 version_file = 'lmdeploy/version.py'
 
 
+def get_backend():
```
Comment (lvhan028) [inline setup.py:11]: How about `get_target_device`

**Comment by grimoire** [comment]
https://github.com/InternLM/lmdeploy/blob/b99a5da255631ea4d24f03bdefccf44bf89aa108/requirements/runtime.txt#L21 We have upgrade torch and triton version.

File: docker/Dockerfile_aarch64_ascend (hunk 1 above)
Comment (jinminxi104) [reply docker/Dockerfile_aarch64_ascend:125]: done

File: setup.py (hunk 2 above)
Comment (jinminxi104) [reply setup.py]: done

**Comment by jinminxi104** [comment]
> https://github.com/InternLM/lmdeploy/blob/b99a5da255631ea4d24f03bdefccf44bf89aa108/requirements/runtime.txt#L21
> 
> We have upgrade torch and triton version.

merged main branch

**Review by grimoire — APPROVED** [review]

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**